src/spectrecoff/Prompt.fs

```diff
@@ -45,6 +52,9 @@ let private prompt prompter =
 let chooseFrom (choices: string list) question = 
     prompt (Prompts.selectionPrompt question choices)
 
+let chooseFromT<'T> (converter : 'T -> string) (choices: 'T list) question =
```

Comment:
While this is the straight forward way to expose the underlying api, we have tried to improve it in several other places as well. I find the `converter` as is not the most user-friendly interace. I suggest creating a type:

```fsharp
type Choice<'A> = 
    { label: string
      value: 'A }
```

and then changing the signature of the prompt function:

```fsharp
chooseFrom (choices: Choice list) question =
````

As you see, this causes a naming conflict with the previously existing method ... I have no good suggestion right now, but I'd not want such a technical name like `chooseFromA`. The obvious one would be `chooseFromChoices` but that sounds a bit silly, too. Or go for a breaking change and use the name for this method, and call the previous one `chooseFromStrings`? Any opinions, @DanielmTheDev ?

Reply:
I wholeheartedly agree to your suggestion about adding a new type, it will make the usage of this function less awkward. 

I'd be more hesitant about that breaking change though, I think something like `chooseFromType` would be perfectly acceptable. Especially considering our [enourmous download numbers](https://www.nuget.org/packages/EluciusFTW.SpectreCoff) now :)

On another note @groma84, maybe you can add an example invocation of this function to the example command to showcase this new functionality?

Reply:
So after a bit more reflection, I definitely agree we should not introduce breaking change. However, I disagree with the name `chooseFromType` because - your not choosing from a type. You're choosing from 'multiple things of a given type'. 

I'd suggest renaming the type `Choice` above to `Option`, and then calling the new function `chooseFromOptions`.

Reply:
I renamed it to `chooseFromValues`, because `Option` already has a (different) meaning in F#

Reply:
@groma84 That is a good point. What about @EluciusFTW s suggestion to get rid of the converter and using a type instead, simplifying to simplify the api somewhat? I think it would be a nice tweak.

Reply:
Okay, will think about it and look into it

Reply:
I have not forgotten about this open PR! I just didn't find the time, yet. Will probably finally be wrapping it up on my side next week

Reply:
@groma84 No worries, thanks for letting us know!

Reply:
I finally found the time to try it out and working with the type made a lot of problems, because the underlying library also expects a converter from 'T to string. But we can't create a converter for 'T because we only have Choice<'T> and would still need to pass a converter for who to get the Label for 'T. Thus I think the original solution is more flexible and easier to understand.

Reply:
Hi @groma84, sorry for the long delay on my side as well. Christmas and all, and also me being busy finishing all the documentation and transforming it to markdown ... 

I had a look at the code myself and added two commits. The last one contains the changes I envisioned with a type encapsulating the label to value mapping. Checkout the example, it's actually rather easy and clean to use imho. A type constraint on the generic (`when 'T: equality`) was needed for the label mapping, but I guess that was needed under the hood also before, and now is made explicit.

Are you fine with it like this? Any comments from you, @DanielmTheDev?

Reply:
Looks good to me!

Reply:
@groma84 Can you please resolve the conflicts? I'd rebase myself, but as its origin is your clone, I can't update the main branch. I recommend you pull main, then rebase this branch onto it and resolve the simple conflict.

Reply:
yes, will do that, but most likely not before Thursday, sorry